core/schema/src/Data/Functor/MaybeLike.hs

```diff
@@ -0,0 +1,14 @@
+module Data.Functor.MaybeLike where
+
+import Data.Functor.Identity
+
+class MaybeLike f where
```

Comment:
Why do we need this new typeclass? 🤔

Reply:
The idea is to give a uniform interface which could work for both `Identity` and `Maybe`. In theory everything could be done using monadic combinators, but by using `likeMaybe` as a view pattern, we can give a nice way to pattern match directly on everything.
Essentially, this tells you: if you can write your server assuming that field may be missing (because they are wrapped in a `MaybeLike` functor), you can make your server work on gRPC, and in the future, as Avro thingies.

Reply:
This caught my eye as well, but your explanation makes sense.